### PR TITLE
src/rgw: HTTP POST object upload

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
Signed-off-by: Kalpesh Pandya <kapandya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/test/rgw/bucket_notification/test_bn.py [hunk 1]
```diff
@@ -1730,6 +1730,50 @@ def test_ps_s3_creation_triggers_on_master_ssl():
...
+
+    conn1 = boto3.client(service_name='s3',aws_access_key_id=get_access_key(),aws_secret_access_key=get_secret_key())
+
+    # create bucket
+    bucket_name = gen_bucket_name()
+    #bucket = conn.create_bucket(bucket_name)
+
+    key_name = 'foo.txt'
+
+    resp = conn1.generate_presigned_post(Bucket=bucket_name, Key=key_name,)
```
Comment (yuvalif) [inline src/test/rgw/bucket_notification/test_bn.py:1759]: you need to create the connection with RGW not the AWS server

Comment (yuvalif) [reply src/test/rgw/bucket_notification/test_bn.py:1759]: see URL in: https://paste.sh/7R2S4FyM#DlbZmQuLlRyw2xdLnr4V2rNa
`https://ruuixb-1.s3.amazonaws.com/`

File: src/test/rgw/bucket_notification/test_bn.py [hunk 2]
```diff
@@ -1730,6 +1730,50 @@ def test_ps_s3_creation_triggers_on_master_ssl():
...
 
+def _get_body(response):
+    body = response['Body']
+    got = body.read()
+    if type(got) is bytes:
+        got = got.decode()
+    return got
+
+
+@attr('sample_test')
```
Comment (TRYTOBE8TME) [inline src/test/rgw/bucket_notification/test_bn.py:1741]: @yuvalif what should be the appropriate attribute used here?

Comment (yuvalif) [reply src/test/rgw/bucket_notification/test_bn.py:1741]: it depends which kind of topic you use: HTTP, kafka or amqp

File: src/test/rgw/bucket_notification/test_bn.py [hunk 3]
```diff
@@ -1730,6 +1730,50 @@ def test_ps_s3_creation_triggers_on_master_ssl():
...
+    url = resp['url']
+
+    payload = OrderedDict([("key" , "foo.txt"),("acl" , "public-read"),\
+    ("Content-Type" , "text/plain"),('file', ('bar'))])
+
+    bucket = conn1.create_bucket(ACL='public-read-write', Bucket=bucket_name)
+
+    r = requests.post(url, files=payload, verify=True)
+    assert_equal(r.status_code, 204)
+    response = conn1.get_object(Bucket=bucket_name, Key='foo.txt')
```
Comment (yuvalif) [inline src/test/rgw/bucket_notification/test_bn.py:1772]: no need to do the "get_object" part
but you should add notifications definitions for POST messages, and verify them

**Pull request merged**